### DISCUSSION TITLE
How should I update the store?

### DISCUSSION THREAD

**Discussion by Wondermarin** [pmndrs/zustand#737] [Q&A] [answered]
Hi!

The documentation shows such an example:
```ts
import create from 'zustand'

const useStore = create(set => ({
  bears: 0,
  increasePopulation: () => set(state => ({ bears: state.bears + 1 })),
  removeAllBears: () => set({ bears: 0 })
}))
```
Where the `increasePopulation` and `removeAllBears` functions do not require copying the store but simply allow you to _merge_ it.

<hr />

However, I found the following code in the [Wiki](https://github.com/pmndrs/zustand/wiki/Updating-nested-state-object-values#normal-approach):
```ts
normalInc: () =>
  set((state) => ({
    ...state,
    deep: {
      ...state.deep,
      nested: {
        ...state.deep.nested,
        obj: {
          ...state.deep.nested.obj,
          count: state.deep.nested.obj.count + 1,
        },
      },
    },
  })),
```
Where the `normalInc` function already copies the store before updating the `deep` object.

<hr />

And now the question itself:
**Can I write such code?**
```ts
normalInc: () =>
  set((state) => ({
    // ...state, w/o this line
    deep: {
      ...state.deep,
      nested: {
        ...state.deep.nested,
        obj: {
          ...state.deep.nested.obj,
          count: state.deep.nested.obj.count + 1,
        },
      },
    },
  })),
```
And why are we copying the store in the Wiki example?

**Comment by dai-shi** [comment, marked as answer]
You are right! That's a mistake in the wiki page.
Although, it's not harmful, let's fix it to avoid confusion. ---done!
